windows/src/engine/keyman32-ver/calldll.cpp

```diff
@@ -575,11 +588,46 @@ LoadDLLs(LPINTKEYBOARDINFO lpkbi) {
 
   if (!GetKeyboardFileName(lpkbi->Name, fullname, _MAX_PATH))
     return FALSE;
-  if ((!lpkbi->lpCoreKeyboard) || (!lpkbi->lpCoreKeyboard)){
+  if (!lpkbi->lpCoreKeyboard){
+    return FALSE;
+  }
+
+  PKEYMAN64THREADDATA _td = ThreadGlobals();
+  if (!_td){
     return FALSE;
   }
 
   km_kbp_keyboard_imx *imx_list = lpkbi->lpIMXList;
+  // return early if the list empty avoiding loading the proxy dll into memory
+  if (!imx_list->library_name) {
+    return FALSE;
+  }
+
+   if (!_td->hModuleProxy) {
+    char module_filename[_MAX_PATH];
+    if(!GetModuleFileName(g_hInstance, module_filename, _MAX_PATH)){
+      SendDebugMessageFormat(0, sdmKeyboard, 0, "LoadDLLsCore: unable to get handle for loaded keyman versioned dll; error:[%d]", GetLastError());
+        return FALSE;
+    }
+
+    char *p = strrchr(module_filename, '\\'); // find last occurrence of '\'
+    if (p != NULL) {
+      size_t len = strnlen(keyman_engine_arch,_MAX_PATH);
```

Comment:
is it possible to use `sizeof(keyman_engine_arch)/sizeof(keyman_engine_arch[0])` here because `keyman_engine_arch` is a const char array?

I also think we should be moving to WCHARs for this code, given it is new code -- so using `GetModuleFileNameW`, `wcsncpy` etc.

Reply:
I can switch to the WCHARs. 

I also had thought that we do have the registry value "root path" that we could read to find where the proxy module is installed. The current way is working on the rule that the versioned dll and proxy will be installed side by side in the same path.

Reply:
> I also had thought that we do have the registry value "root path" that we could read to find where the proxy module is installed. The current way is working on the rule that the versioned dll and proxy will be installed side by side in the same path.

Yes, I think side-by-side is safer than relying on registry. Also allows for development-time work more easily.